### PR TITLE
Api 2731 unknown vista error handling

### PR DESCRIPTION
**Pull request by ilaflamme-va**
### WAIT TO MERGE UNTIL AFTER https://github.com/department-of-veterans-affairs/lighthouse-vistalink/pull/18

### CODE REVIEW

File: vistalink/src/main/java/gov/va/api/lighthouse/vistalink/service/controller/ParallelRpcExecutor.java [hunk 1]
```diff
@@ -57,6 +76,9 @@ public RpcResponse execute(RpcRequest request) {
 
   @SneakyThrows
   private RpcInvocationResult failed(String vista, String message) {
+    if (message.contains("VistaLoginModuleTooManyInvalidAttemptsException")) {
```
Comment (bschofield-va) [inline vistalink/src/main/java/gov/va/api/lighthouse/vistalink/service/controller/ParallelRpcExecutor.java:79]: no bueno merge

File: vistalink/src/main/java/gov/va/api/lighthouse/vistalink/service/controller/VistaLinkExceptions.java [hunk 2]
```diff
@@ -0,0 +1,24 @@
+package gov.va.api.lighthouse.vistalink.service.controller;
```
Comment (bschofield-va) [inline vistalink/src/main/java/gov/va/api/lighthouse/vistalink/service/controller/VistaLinkExceptions.java:1]: no bueno merge

File: vistalink/src/main/java/gov/va/api/lighthouse/vistalink/service/controller/WebExceptionHandler.java [hunk 3]
```diff
@@ -36,6 +38,12 @@ public RpcResponse handleBadRequestBody(Exception e, HttpServletRequest request)
     return failedResponseFor("Failed to read request body.");
   }
 
+  @ExceptionHandler({UnknownVista.class})
+  @ResponseStatus(HttpStatus.BAD_REQUEST)
+  public RpcResponse handleUnknownVistaIncludes(Exception e, HttpServletRequest request) {
+    return failedResponseFor("Unknown vista site specified in request includes[].");
```
Comment (bschofield-va) [inline vistalink/src/main/java/gov/va/api/lighthouse/vistalink/service/controller/WebExceptionHandler.java:44]: what's with the `[]` at the end of the message? Is that mean to indicate a list? Eitherway, remove it and just say _... request includes value_.

**Comment by ilaflamme-va** [comment]
Merge fixed.

**Review by bschofield-va — CHANGES_REQUESTED** [review]
don't merge this yet... I'm pretty sure this doesn't work quite right

File: vistalink/src/main/java/gov/va/api/lighthouse/vistalink/service/controller/ParallelRpcExecutor.java [hunk 4]
```diff
@@ -20,26 +21,43 @@
...
+  }
+
   @Override
   public RpcResponse execute(RpcRequest request) {
     var response = RpcResponse.builder();
     List<ConnectionDetails> targets = vistaNameResolver.resolve(request.target());
     if (targets.isEmpty()) {
       return response.status(Status.NO_VISTAS_RESOLVED).build();
     }
+    checkRequestVistasAreKnown(targets, request);
```
Comment (bschofield-va) [inline vistalink/src/main/java/gov/va/api/lighthouse/vistalink/service/controller/ParallelRpcExecutor.java]: The name resolve has _already found it_. I think this is the wrong place to be asking this question. That is, it's not the executor responsibility to check whether a name exists or not... it's the name resolvers.

**Comment by bschofield-va** [comment]
Can be tested with:
```
$ ./invoke-rpc requests/unknown-vista-error.json -w "\n%{http_code}\n"
{"status":"FAILED","message":"Unknown vista site specified: [nope]","results":[]}
400
```

**Review by bschofield-va — APPROVED** [review]

**Pull request merged**